pkg/internal/cryptoutil/utils.go

```diff
@@ -49,6 +68,8 @@ func IsChachaKeyValid(key []byte) bool {
 
 // KeyPair represents a private/public key pair
 type KeyPair struct {
+	// Type is the type of keys enclosed in this key pair
+	Type string
```

Comment:
Why string?

Reply:
because it could be either a signature or encryption key type, we can't enforce struct type in this case

Reply:
or we create two KeyPair types  and remove Type altogether